src/components/subsetDraft/SubsetMetadata.jsx

```diff
@@ -1,95 +1,152 @@
-                                className='datepicker'/>
+            <section style={{ margin: "5px 0 5px 0" }}>
+                <div
+                    style={{ float: "left", marginRight: "20px", padding: "0" }}
+                >
+                    <label
+                        style={{
+                            display: "block",
+                            fontSize: "16px",
+                            fontFamily: "Roboto"
+                        }}
+                        htmlFor='from_date'
+                    >
+                        {t("Valid from")}:{" "}
+                    </label>
+                    <DatePicker
+                        id='from_date'
+                        style={{ display: "block" }}
+                        value={subset.draft.validFrom}
+                        onChange={date => handleDateChange(date, 'from')}
+                        clearIcon={null}
+                        format='dd.MM.y'
+                        locale={languages.find(i => i.default).IETF}
+                        className='datepicker'
+                    />
                 </div>
 
-                <div style={{float: 'left'}}>
-                    <label style={{display: 'block', fontSize: '16px', fontFamily: 'Roboto'}}
-                           htmlFor="to_date">{t('Valid to')}: </label>
-                    <DatePicker id='to_date' style={{display: 'block'}}
-                                value={subset.draft.validUntil}
-                                onChange={(date) => subset.dispatch({action: 'to', data: date})}
-                                clearIcon={null}
-                                format='dd.MM.y'
-                                locale={languages.find(i => i.default).IETF}
-                                className='datepicker'/>
+                <div style={{ float: "left" }}>
+                    <label
+                        style={{
+                            display: "block",
+                            fontSize: "16px",
+                            fontFamily: "Roboto"
+                        }}
+                        htmlFor='to_date'
+                    >
+                        {t("Valid to")}:{" "}
+                    </label>
+                    <DatePicker
+                        id='to_date'
+                        style={{ display: "block" }}
+                        value={subset.draft.validUntil}
+                        onChange={date => handleDateChange(date,'to')}
+                        clearIcon={null}
+                        format='dd.MM.y'
+                        locale={languages.find(i => i.default).IETF}
+                        className='datepicker'
+                    />
                 </div>
-                <br style={{clear: 'both'}}/>
+                <br style={{ clear: "both" }} />
+
+                <ChangeDateModal isShowing={isShowing} submit={submit} dismiss={dismiss}/>
```

Comment:
This is a component independent from datepicker activities, it listents to AppContext's meaning about the date provided. This meaning could be expressed in some varible ('doWarn'), which is normally is null, but in cases when the provided user input satisfices or not satisfices some criteria, will be set to some value ({message: 'Attention: you're gouing to reset all codes', header: 'Change date warningHeader'...}).

`(subset.draft.validFrom.doWarn || subset.draft.valifUntil.doWarn) && <ChangeDateModal />`
The rest of the handling of the warning and the options presented to the user should be incapsulateded into the modal. When user decides, modal dies. The datepicker still knows nothing about this activities, it just gets new dates from the state.
Low coupling.

This is going to be a pattern for the error and warning handling. 

Yet the error and warning handling is not implemented in AppContext. You will be the first. 
Each form field has its own restrictions and rules. Each form input component has to have a warning modal and and some error reaction style whith message. All restrictions, criteria and dependencies between the filds will be gathered in the useSubset hook in the AppContext. AppContesxt.subset will signal about the errors and emit warning by changing special field related variables in the state. The field modals and styling options will listen to the state variables specified for theirs fields and conditionally react on them or stay quiet when they are nulls.